app/models/api_action.rb

```diff
@@ -0,0 +1,78 @@
+class ApiAction < ApplicationRecord
+  include Encryptable
+  attr_encrypted :bearer_token
+
+  belongs_to :api_namespace, optional: true
+  belongs_to :api_resource, optional: true
+
+  enum action_type: { send_email: 0, send_web_request: 1, redirect: 2, serve_file: 3 }
+
+  enum lifecycle_stage: {initialized: 0, executing: 1, complete: 2, failed: 3, discarded: 4}
+
+  default_scope { order(position: 'ASC') }
+
+  ransacker :action_type, formatter: proc {|v| action_types[v]}
+
+  has_rich_text :custom_message
+
+  def self.children
+    ['new_api_actions', 'create_api_actions', 'show_api_actions', 'update_api_actions', 'destroy_api_actions', 'error_api_actions']
+  end
+
+  def execute_action
+    self.update(lifecycle_stage: 'executing')
+    send(action_type)
+  end
+
+  private
+
+  def send_email
+    begin
+      ApiActionMailer.send_email(self).deliver_now
+      self.update(lifecycle_stage: 'complete', lifecycle_message: email)
+    rescue => e
+      self.update(lifecycle_stage: 'failed', lifecycle_message: e.message)
+      execute_error_actions
+    end
+  end
+
+  def send_web_request
+    begin
+      response = HTTParty.post(request_url, 
```

Comment:
@Pralish This should be settable, see https://github.com/restarone/violet_rails/issues/258